src/Lokad.Tokenizers/Tokenizer/TokenizationUtils.cs

```diff
@@ -410,7 +410,7 @@ public static void DecomposeNfkc(Token token)
 
             //HINT: [@eslam] check if character is removed from the original text after normalization
             if (!token.Text.EnumerateRunes().Contains(character))
-                extraCharSize = -1;
+                extraCharSize -= character.Utf8SequenceLength - character.Utf16SequenceLength;// -1;
```

Comment:
I think it would be better to have a nkfc enumerator that returns the characters and the extra characters count values to avoid any additional processing (https://github.com/guillaume-be/rust-tokenizers/blob/main/main/src/tokenizer/tokenization_utils.rs#L215)

Reply:
Agree, PR updated.